hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+import com.uber.hoodie.common.table.log.block.HoodieAvroDataBlock;
+import com.uber.hoodie.common.table.log.block.HoodieCorruptBlock;
+import com.uber.hoodie.common.table.log.block.HoodieLogBlock;
+import com.uber.hoodie.common.util.FSUtils;
+import com.uber.hoodie.config.HoodieCompactionConfig;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.status.api.v1.NotFoundException;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.stereotype.Component;
+import parquet.avro.AvroSchemaConverter;
+import parquet.schema.MessageType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class HoodieLogFileCommand implements CommandMarker {
+
+  @CliAvailabilityIndicator({"show logfiles"})
+  public boolean isShowArchivedLogFileAvailable() {
+    return HoodieCLI.tableMetadata != null;
+  }
+
+  @CliCommand(value = "show logfile commits", help = "Read commits from log files")
+  public String showLogFileCommits(
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified path for the log file")
+      final String logFilePathPattern) throws IOException {
+
+
+    List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+        HoodieCLI.tableMetadata.getHadoopConf()).globStatus(new Path(logFilePathPattern)))
+        .map(status -> status.getPath().toString()).collect(Collectors.toList());
+    Map<String, Integer> commitToCount = Maps.newHashMap();
+
+    for(String logFilePath: logFilePaths) {
+      FileStatus[] fsStatus = FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()).listStatus(
+          new Path(logFilePath));
+      Schema writerSchema = new AvroSchemaConverter().convert(readSchemaFromLogFile(new Path(logFilePath)));
+      HoodieLogFormat.Reader reader = HoodieLogFormat.newReader(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()),
+          new HoodieLogFile(fsStatus[0].getPath()), writerSchema, true);
+
+      // read the avro blocks
+      while (reader.hasNext()) {
+        HoodieLogBlock n = reader.next();
+        String commitTime;
+        if (n instanceof HoodieCorruptBlock) {
+          try {
+            commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+          } catch (Exception e) {
+            commitTime = "corrupt_block_1";
```

Comment:
why is it always `corrupt_block_1` ? did you intend to bump up a counter for each corrupt block?

Reply:
yeah, counter thing, done.